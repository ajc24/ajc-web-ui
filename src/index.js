/* Export all of the header components */
export { default as Header } from './ui-elements/header/Header';

/* Export all of the icon components */
export { default as SmallDownArrowIcon } from './ui-elements/icons/SmallDownArrowIcon';
export { default as SmallRightArrowIcon } from './ui-elements/icons/SmallRightArrowIcon';
