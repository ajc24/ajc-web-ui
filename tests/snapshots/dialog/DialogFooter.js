import 'jsdom-global/register';
import React from 'react';
import { TestDev } from 'ajc-jest-enzyme';
import { DialogFooter } from '../../../src';

describe('DialogFooter', () => {
  const twoButtonsList = [
    {
      id: 'save-button',
      onClick: () => {},
      title: 'Save',
      type: 'submit',
    },
    {
      id: 'cancel-button',
      onClick: () => {},
      title: 'Cancel',
      type: 'button',
    }
  ];

  describe('Default props and rendering - Component with grey background', () => {
    let jsonSnapshot;

    beforeAll(() => {
      jsonSnapshot = TestDev.createSnapshot(
        <div role="region">
          <DialogFooter id="test-dialog-buttons-list-id" />
        </div>
      );
    });

    it('verifies the snapshot for the component', () => {
      expect(jsonSnapshot).toMatchSnapshot();
    });
  });

  describe('Transferred props and rendering - Component with red background', () => {
    let jsonSnapshot;

    beforeAll(() => {
      jsonSnapshot = TestDev.createSnapshot(
        <div role="region">
          <DialogFooter id="test-dialog-buttons-list-id" />
        </div>
      );
    });

    it('verifies the snapshot for the component', () => {
      expect(jsonSnapshot).toMatchSnapshot();
    });
  });

  describe('Transferred props and rendering - Grey component with buttons list', () => {
    let jsonSnapshot;

    beforeAll(() => {
      jsonSnapshot = TestDev.createSnapshot(
        <div role="region">
          <DialogFooter id="test-dialog-buttons-list-id" colour="grey" buttonsList={twoButtonsList} />
        </div>
      );
    });

    it('verifies the snapshot for the component', () => {
      expect(jsonSnapshot).toMatchSnapshot();
    });
  });

  describe('Transferred props and rendering - Red component with buttons list', () => {
    let jsonSnapshot;

    beforeAll(() => {
      jsonSnapshot = TestDev.createSnapshot(
        <div role="region">
          <DialogFooter id="test-dialog-buttons-list-id" colour="red" buttonsList={twoButtonsList} />
        </div>
      );
    });

    it('verifies the snapshot for the component', () => {
      expect(jsonSnapshot).toMatchSnapshot();
    });
  });
});
