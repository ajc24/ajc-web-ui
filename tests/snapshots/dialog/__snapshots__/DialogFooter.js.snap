// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DialogFooter Default props and rendering - Component with grey background verifies the snapshot for the component 1`] = `
<div
  role="region"
>
  <div
    className="ajc-flex-row ajc-flex-justify-center ajc-dialog-footer ajc-width-dialog ajc-font-default ajc-background-grey-2"
    id="test-dialog-buttons-list-id"
  />
</div>
`;

exports[`DialogFooter Transferred props and rendering - Component with red background verifies the snapshot for the component 1`] = `
<div
  role="region"
>
  <div
    className="ajc-flex-row ajc-flex-justify-center ajc-dialog-footer ajc-width-dialog ajc-font-default ajc-background-grey-2"
    id="test-dialog-buttons-list-id"
  />
</div>
`;

exports[`DialogFooter Transferred props and rendering - Grey component with buttons list verifies the snapshot for the component 1`] = `
<div
  role="region"
>
  <div
    className="ajc-flex-row ajc-flex-justify-center ajc-dialog-footer ajc-width-dialog ajc-font-default ajc-background-grey-2"
    id="test-dialog-buttons-list-id"
  >
    <button
      className="ajc-dialog-footer-button ajc-text-white ajc-menu-item-grey"
      id="save-button"
      onClick={[Function]}
      type="submit"
    >
      Save
    </button>
    <button
      className="ajc-dialog-footer-button ajc-text-white ajc-menu-item-grey"
      id="cancel-button"
      onClick={[Function]}
      type="button"
    >
      Cancel
    </button>
  </div>
</div>
`;

exports[`DialogFooter Transferred props and rendering - Red component with buttons list verifies the snapshot for the component 1`] = `
<div
  role="region"
>
  <div
    className="ajc-flex-row ajc-flex-justify-center ajc-dialog-footer ajc-width-dialog ajc-font-default ajc-background-red"
    id="test-dialog-buttons-list-id"
  >
    <button
      className="ajc-dialog-footer-button ajc-text-white ajc-menu-item-red"
      id="save-button"
      onClick={[Function]}
      type="submit"
    >
      Save
    </button>
    <button
      className="ajc-dialog-footer-button ajc-text-white ajc-menu-item-red"
      id="cancel-button"
      onClick={[Function]}
      type="button"
    >
      Cancel
    </button>
  </div>
</div>
`;
